### PR TITLE
Model api integration test + more debugs

### PR DESCRIPTION
**Pull request by mwielgus**
Depends on #554.

### CODE REVIEW

**Comment by mwielgus** [comment]
cc: @vishh @piosz

**Comment by mwielgus** [comment]
Fixes #487

**Comment by mvdan** [comment]
Note that this collides with #554

File: integration/heapster_api_test.go [hunk 1]
```diff
@@ -190,6 +207,31 @@ func getSchema(fm kubeFramework, svc *kube_api.Service) (*api_v1.TimeseriesSchem
...
+		Prefix("proxy").
+		Resource("services").
+		Name(svc.Name).
+		Suffix(url).
+		Do().Raw()
+	if err != nil {
+		return nil, err
+	}
+	var metrics api_v1.MetricResultList
+	if err := json.Unmarshal(body, &metrics); err != nil {
```
Comment (vishh) [inline integration/heapster_api_test.go]: Do we want to verify that all the expected resources are present in the response?

File: integration/heapster_api_test.go [hunk 2]
```diff
@@ -190,6 +207,31 @@ func getSchema(fm kubeFramework, svc *kube_api.Service) (*api_v1.TimeseriesSchem
 	return &timeseriesSchema, nil
 }
 
+func getModelMetrics(fm kubeFramework, svc *kube_api.Service, pod *kube_api.Pod) (*api_v1.MetricResultList, error) {
+	url := fmt.Sprintf("/api/v1/model/namespaces/%s/pod-list/%s/metrics/%s",
```
Comment (vishh) [inline integration/heapster_api_test.go]: What about all the other model APIs?

Comment (mvdan) [inline integration/heapster_api_test.go]: Maybe the issue should be closed once all endpoints are tested. Is that what you mean?

Comment (vishh) [inline integration/heapster_api_test.go]: Yeah. That makes sense.

File: integration/heapster_api_test.go (hunk 1 above)
Comment (mwielgus) [inline integration/heapster_api_test.go]: Added more checks.

File: integration/heapster_api_test.go (hunk 2 above)
Comment (mwielgus) [inline integration/heapster_api_test.go]: I would rather check the other endpoints in a more mocked and unit-like test for `api/v1/model_handler.go` where I can verify if the returned values are correct.

Comment (vishh) [inline integration/heapster_api_test.go]: Would unit-like tests validate the restful layer as well? If not, at-least
poking those endpoints to make sure they work will be useful.

Comment (mwielgus) [inline integration/heapster_api_test.go]: Well, I hope so. I can still poke all of the endpoints and check if they don't crash (but probably in the next PR).

Comment (mwielgus) [inline integration/heapster_api_test.go]: With the E2E approach as here i can only check if the output is returned at all and if it is sane, not if the endpoints are correctly bound to model funcs.

Comment (vishh) [inline integration/heapster_api_test.go]: Fair point! I see value for all of them and in terms of priority, the
bounding of methods might be the final thing we want to test, with the
assumption that code reviews would have caught that.

Comment (vishh) [inline integration/heapster_api_test.go]: I am perfectly fine with adding checks for the other endpoints in a future
PR. I just wanted to bring that up.

File: integration/heapster_api_test.go [hunk 3]
```diff
@@ -431,13 +530,16 @@ func apiTest(kubeVersion string) error {
...
 			break
 		}
 		if attempts == 0 {
+			glog.V(2).Info("Too many attempts.")
 			return err
 		}
+		glog.V(2).Infof("Some tests failed. Retrying.")
 		attempts--
-		time.Sleep(time.Second)
+		time.Sleep(time.Second * 10)
```
Comment (vishh) [inline integration/heapster_api_test.go]: This change might cause timeouts in the `go test` framework. We will have to update the timeout in the makefiles.

**Comment by vishh** [comment]
@mwielgus: Just one more comment. Otherwise LGTM. This is super useful!

File: integration/heapster_api_test.go (hunk 3 above)
Comment (mwielgus) [inline integration/heapster_api_test.go]: If this test doesn't complete in <10min then something is terribly broken (or your connection is dead slow). Increasing the timeout beyond 30m will not make any good.

Comment (vishh) [inline integration/heapster_api_test.go]: Fair point. Lets go with this for now.

**Pull request merged**